libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1323,6 +1323,23 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		}
 		break;
 	}
+	case FunctionType::Kind::BytesConcat:
+	{
+		TypePointers argumentTypes;
+		vector<string> argumentVars;
+		for (ASTPointer<Expression const> const& argument: arguments)
+		{
+			argumentTypes.emplace_back(&type(*argument));
+			argumentVars += IRVariable(*argument).stackSlots();
```

Comment:
Is the total number of stack slots always same as the number of arguments here? Can we add this assert?

Reply:
It's not, but since we pass the vector of types, this is exactly how it is done. Maybe I misundertood, though...

Reply:
After the for loop, would `_argumentVars.size() == arguments.size()` be true?

Reply:
Actually it will not be. This is because of for example `bytes calldata` which takes 2 slots on stack